src/Form/Field.php

```diff
@@ -255,6 +255,30 @@ public function isRequired(): bool
         return $this->required ?? false;
     }
 
+    protected function isSaveable(): bool
+    {
+        if ($this->isRequired() === true && $this->save() === true && $this->isEmpty() === true) {
+            // check the data of the relevant fields if there is a when option
+            if (empty($this->when) === false && is_array($this->when) === true) {
+                $input = $this->attrs['input'] ?? [];
+
+                if (empty($input) === false) {
+                    foreach ($this->when as $field => $value) {
+                        // if the input data doesn't have data for when fields
+                        // or doesn't match, that means is not required and return true
+                        if (($input[$field] ?? '') !== $value) {
+                            return true;
```

Comment:
Doesn't this early return make this comparison an `OR` comparison (= only one of the `when` fields needs to match)? Currently our `when` conditions are `AND` conditions, aren't they?

Reply:
Yes, `when` the option works with the AND operator.

This line means that if one of conditions doesn't match, field is not required and can be saved.

Reply:
Right, I forgot to think around two corners here. 😄

Could you add an explicit comment about this to the code to make it obvious when we read this code in the future?